apps/queries/api.py

```diff
@@ -203,83 +203,150 @@ def get(self, request, **kwargs):
+        bc.f_name_dutch as bc_f_name_dutch, bc.f_name_english as bc_f_name_english, bc.f_entity_id as bc_f_entity_id, 
+        bc.ins_entity_id as bc_ins_entity_id, bc.ins_name_dutch as bc_ins_name_dutch, bc.ins_name_english as bc_ins_name_english
+    FROM users u INNER JOIN institution_institution institution ON u.institution_id = institution.id
+    LEFT JOIN (
+        SELECT st_in.may_update, st_in.user_id, st_in.institution_id, 
+            ins.name_english, ins.name_dutch, ins.entity_id
+        FROM staff_institutionstaff st_in
+            INNER JOIN institution_institution ins ON st_in.institution_id = ins.id
+    ) ins ON (u.id = ins.user_id)
+    LEFT JOIN (
+        SELECT st_fa.may_update, st_fa.may_award, st_fa.may_administrate_users, st_fa.user_id, st_fa.faculty_id,
+            fac.name_english, fac.name_dutch, fac.entity_id,
+            ins.name_english as ins_name_english, ins.name_dutch as ins_name_dutch, ins.entity_id as ins_entity_id
+        FROM staff_facultystaff st_fa
+            INNER JOIN institution_faculty fac ON st_fa.faculty_id = fac.id
+            INNER JOIN institution_institution ins ON fac.institution_id = ins.id            
+    ) f ON (u.id = f.user_id)
+    LEFT JOIN (
+        SELECT st_is.may_update, st_is.may_award, st_is.may_administrate_users, st_is.user_id, st_is.issuer_id,
+            iss_iss.name_english, iss_iss.name_dutch, iss_iss.entity_id,
+            fac.name_english as f_name_english, fac.name_dutch as f_name_dutch, fac.entity_id as f_entity_id,
+            ins.name_english as ins_name_english, ins.name_dutch as ins_name_dutch, ins.entity_id as ins_entity_id
+        FROM staff_issuerstaff st_is
+            INNER JOIN issuer_issuer iss_iss ON iss_iss.id = st_is.issuer_id 
+            INNER JOIN institution_faculty fac ON iss_iss.faculty_id = fac.id
+            INNER JOIN institution_institution ins ON fac.institution_id = ins.id
+    ) i ON (u.id = i.user_id)
+    LEFT JOIN (
+        SELECT st_bc.may_update, st_bc.may_award, st_bc.may_administrate_users, st_bc.user_id, st_bc.badgeclass_id,
+            ib.name, ib.entity_id,
+            iss_iss.name_english as i_name_english, iss_iss.name_dutch as i_name_dutch, iss_iss.entity_id as i_entity_id,
+            fac.name_english as f_name_english, fac.name_dutch as f_name_dutch, fac.entity_id as f_entity_id,
+            ins.name_english as ins_name_english, ins.name_dutch as ins_name_dutch, ins.entity_id as ins_entity_id
+        FROM staff_badgeclassstaff st_bc  
+            INNER JOIN issuer_badgeclass ib ON ib.id = st_bc.badgeclass_id
+            INNER JOIN issuer_issuer iss_iss ON iss_iss.id = ib.issuer_id 
+            INNER JOIN institution_faculty fac ON iss_iss.faculty_id = fac.id
+            INNER JOIN institution_institution ins ON fac.institution_id = ins.id
+    ) bc ON (u.id = bc.user_id)"""
+            if filter_by_institution:
```

Comment:
This if could also be removed and query rewritten with something like: "u.institution_id = IFNULL(%(ins_id)s, u.institution_id)" (but not mandatory ..)

Reply:
https://github.com/edubadges/edubadges-server/pull/172/commits/24644833e9c7e9145bc5321569bc448946001b76